### PR TITLE
Detmatch bugfix and from_ctx constructor

### PR DESCRIPTION
**Pull request by jlashner**
Bugfix from last commit of the det-match PR.

Also adds a utility function that can be used to load a resonator set from a context that contains det_cal metadata.

### CODE REVIEW

File: sotodlib/coords/det_match.py [hunk 1]
```diff
@@ -167,6 +179,54 @@ def as_array(self):
...
+        Args
+        ----------
+        ctx: Context
+            Context object
+        obs_id: str
+            Obs id to base ResSet off
+        stream_id: str
+            Stream id for ResSet to load
+        cal_name: str
+            Attribute of the axismanager that holds detector calibration info
```
Comment (mhasself) [inline sotodlib/coords/det_match.py:196]: [As per this](https://github.com/simonsobs/sotodlib/issues/79), a more "sotodlib" way to do this is to have an argument that is the same name as the default field within am, and default it to None.  I.e. `det_cal=None`.  The None will trigger the default to be looked up in `am`; if it isn't None then `det_cal` will be used in place of am[det_cal].  An optional extension to this behavior is to also detect when det_cal is a string and use det_cal = am[det_cal] in that case.

I won't insist on this change, here... depends on how much you see it getting used, how flexible you want to be.

File: sotodlib/coords/det_match.py [hunk 2]
```diff
@@ -167,6 +179,54 @@ def as_array(self):
             data.append(tuple(getattr(r, name) for name, _ in dtype))
         return np.array(data, dtype=dtype)
 
+    @classmethod
+    def from_ctx(cls, ctx, obs_id, stream_id, cal_name='det_cal', am=None):
```
Comment (mhasself) [inline sotodlib/coords/det_match.py:183]: "aman" is a more standard arg name than `am` here.

**Review by mhasself — APPROVED** [review]
I have no serious issue with this but see comments about arg conventions.

**Comment by jlashner** [comment]
Made changes based on @mhasself's suggestions, and replaced `from_ctx` to `from_aman`, which is more generally applicable.

**Pull request merged**